### DISCUSSION TITLE
Some keybindings don't work as defined

### DISCUSSION THREAD

**Discussion by mustaqimM** [meow-edit/meow#472] [Q&A]
Hello, I commented out my .emacs.d folder and I copied the default keybindings from https://github.com/meow-edit/meow/blob/master/KEYBINDING_QWERTY.org

I have some problems:

* `a`: `meow-append` doesn't append to the right of the cursor. It works as `i`. It is mapped to `meow-append` and calling it directly has the same issue.
* `w` and `W` work the same: `W` doesn't select the whitespace. For a word like `doesn't`, both select up to `doesn` and doesn't seem different to `e`.
* Another is why does `d` and `s` perform virtually the same function?
* I can't repeatedly press `w` to move forward, but I think that's intentional. But why? Because in helix `e` and `w` work in a similar manner. Here `w` is just for the single word (to work on).
* In vim I can say `vi(` to select the next (), like `(1873-1877)`, but I can't here unless I'm actually inside the ().
* `t` and `f` both jump to the given character. `t` moves to just before the character.

Also, why does capital `I` and `A` mean insert above when the lowercase ones mean insert before and after?

Would there be any issues if I remap to the helix keybindings? And thank you for this project! Beacon mode is great!

**Comment by zetashift** [comment]
I sadly can't answer all of them, but I can answer some of them!

> a: meow-append doesn't append to the right of the cursor. It works as i. It is mapped to meow-append and calling it directly has the same issue.


According to the FAQ:
`In Emacs cursor is located between two characters, so we would have to implement a hack to change the behaviour. You can read more about it in https://github.com/meow-edit/meow/discussions/87.`
See https://github.com/meow-edit/meow/blob/master/FAQ.org#frequently-asked-questions

> Also, why does capital I and A mean insert above when the lowercase ones mean insert before and after?

Was also quite surprising to me, I remapped.

> Would there be any issues if I remap to the helix keybindings? And thank you for this project! Beacon mode is great!

I don't think it's any problem or thing if you remap as you like. That's the beauty of emacs.
However I tried this as well, and it works all fine until you're messing with other modes like `magit` and `lsp`. 
If there was a config example that kinda shows how to transition from Helix/Kakoune -> `meow` that would be awesome,but until then it's gonna be finding the right balance between how meow makes emacs keybindings easier with modal editing and helix/kakoune keybindings.

**Comment by DogLooksGood** [comment]
This project is inspired by kak/helix, however it has a different goal. One goal is to reduce the usages of modifier keys. So it doesn't work like kak/helix, due to the way how meow handles the backward direction and selection extending.

**Comment by Linerre** [comment]
> * In vim I can say `vi(` to select the next (), like `(1873-1877)`, but I can't here unless I'm actually inside the ().

With meow, I usually press `o` to select the next block, be it a `(sth)` or `{sth}`. I think this is neat and nicer than Vim's way.

```rust
 fn eq(&self, other: &Self) -> bool {
 //^ point at here to the left of the first `(', press o
 // point can be at any place before the left `(' in this case
     self.name == other.name && self.age == other.age
    }
```

The whole `(&self, other: &Self)` will be selected.

**Comment by asymmetric** [comment]
> I can't repeatedly press w to move forward, but I think that's intentional. But why? Because in helix e and w work in a similar manner. Here w is just for the single word (to work on).

I also found this confusing, coming from both Vim and Helix. I thought of swapping the functions attached to `w` and `e` as a solution, but haven't done it yet.
